### PR TITLE
[v6r22] Update of index function and addition of update function in ElasticSearchDB.py

### PR DESCRIPTION
**Pull request by panta-123**
BEGINRELEASENOTES

*ElasticSearchDB

Change: Existing index function is modified to add argument option of document id 
NEW:  update function is added.

ENDRELEASENOTES

### CODE REVIEW

**Comment by panta-123** [comment]
A patch relase of v6r22, to use in Belle II experiment. This is to test new feature using Elasticsearch .

File: Core/Utilities/ElasticSearchDB.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
 from DIRAC.FrameworkSystem.Client.BundleDeliveryClient import BundleDeliveryClient
 
 
+sLog = gLogger.getSubLogger(__name__)
```
Comment (fstagni) [inline Core/Utilities/ElasticSearchDB.py:24]: If you are using this sublogger (why not, I like it) then please apply to all the module. It's just replacing `gLogger.` with `sLog.`.

Just one minor comment: we normally call it `LOG`:
```suggestion
LOG = gLogger.getSubLogger(__name__)
```

Comment (fstagni) [reply Core/Utilities/ElasticSearchDB.py:24]: Which is, BTW, what's done in the upper branches.

Comment (andresailer) [reply Core/Utilities/ElasticSearchDB.py:24]: @atsareg requested that we move to `sLog` https://github.com/DIRACGrid/DIRAC/pull/4304#discussion_r334868828, and we did https://github.com/DIRACGrid/DIRAC/commit/033b527fc429c75f0834fc1c9223211f61899e8b

Comment (fstagni) [reply Core/Utilities/ElasticSearchDB.py:24]: Ah, OK. Good then.

Comment (panta-123) [reply Core/Utilities/ElasticSearchDB.py:24]: I have repleaced all gLogger with sLog in recent commit.

File: Core/Utilities/ElasticSearchDB.py [hunk 2]
```diff
@@ -110,6 +112,30 @@ def query(self, index, query):
     except RequestError as re:
       return S_ERROR(re)
 
+  def update(self, index, doctype='_doc', query=None, updateByQuery=True, id=None):
+    """ Executes an update of a document, and returns S_OK/S_ERROR
+    :param self: self reference
```
[the remark this answers is not in the document]
Comment (panta-123) [reply Core/Utilities/ElasticSearchDB.py:118]: The has been resolved in recent commit.

File: Core/Utilities/ElasticSearchDB.py (hunk 1 above)
Comment (chaen) [reply Core/Utilities/ElasticSearchDB.py:24]: we did move to `sLog`, totally ignoring my comment and our own rules :D

Comment (chaen) [reply Core/Utilities/ElasticSearchDB.py:24]: more seriously, if we want to make t hat a (good) habit, let's put it somewhere in the developer guide once for all

Comment (panta-123) [reply Core/Utilities/ElasticSearchDB.py:24]: @atsareg , Is there anymore stuff that needs to be done ? 
If not then we can merge this.

**Pull request merged**